ingestion/src/metadata/utils/secrets/manage_ssl.py

```diff
@@ -0,0 +1,166 @@
+    DorisConnection,
+)
+from metadata.generated.schema.entity.services.connections.database.greenplumConnection import (
+    GreenplumConnection,
+)
+from metadata.generated.schema.entity.services.connections.database.mysqlConnection import (
+    MysqlConnection,
+)
+from metadata.generated.schema.entity.services.connections.database.postgresConnection import (
+    PostgresConnection,
+)
+from metadata.generated.schema.entity.services.connections.database.redshiftConnection import (
+    RedshiftConnection,
+)
+from metadata.generated.schema.entity.services.connections.messaging.kafkaConnection import (
+    KafkaConnection,
+)
+from metadata.generated.schema.security.ssl import verifySSLConfig
+from metadata.ingestion.connections.builders import init_empty_connection_arguments
+
+
+class SSLManager:
+    "SSL Manager to manage SSL certificates for service connections"
+
+    def __init__(self, ca, key=None, cert=None):
+        self.temp_files = []
+        self.ca_file_path = self.create_temp_file(ca)
+        self.cert_file_path = None
+        self.key_file_path = None
+        if cert:
+            self.cert_file_path = self.create_temp_file(cert)
+        if key:
+            self.key_file_path = self.create_temp_file(key)
+
+    def create_temp_file(self, content):
+        with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+            temp_file.write(content.get_secret_value().encode())
+            temp_file.close()
+        self.temp_files.append(temp_file.name)
+        return temp_file.name
+
+    def cleanup_temp_files(self):
+        for temp_file in self.temp_files:
+            try:
+                os.remove(temp_file)
+            except FileNotFoundError:
+                pass
+        self.temp_files = []
+
+    @singledispatchmethod
+    def setup_ssl(self, connection):
+        raise NotImplementedError(f"Connection {type(connection)} type not supported")
+
+    @setup_ssl.register(MysqlConnection)
+    @setup_ssl.register(DorisConnection)
+    def _(self, connection):
+        # Use the temporary file paths for SSL configuration
+        connection = cast(Union[MysqlConnection, DorisConnection], connection)
+        connection.connectionArguments = (
+            connection.connectionArguments or init_empty_connection_arguments()
+        )
+        ssl_args = connection.connectionArguments.__root__.get("ssl", {})
+        if connection.sslConfig.__root__.caCertificate:
+            ssl_args["ssl_ca"] = self.ca_file_path
+        if connection.sslConfig.__root__.sslCertificate:
+            ssl_args["ssl_cert"] = self.cert_file_path
+        if connection.sslConfig.__root__.sslKey:
+            ssl_args["ssl_key"] = self.key_file_path
+        connection.connectionArguments.__root__["ssl"] = ssl_args
+        return connection
+
+    @setup_ssl.register(PostgresConnection)
+    @setup_ssl.register(RedshiftConnection)
+    @setup_ssl.register(GreenplumConnection)
+    def _(self, connection):
```

Comment:
is the benefit of having these defined here vs. in the `connection.py` for each connector the duplication? of code? It might be easier to maintain/find if we have it grouped with each connector 🤔

Reply:
the original thought was single place for all SSL related things, this similar approach we use in profiler to register different connection types to different logics for metrics. one of the reasons to have added the same is the sonar cloud complaining about duplication, and 2nd would be initialisation of ssl_manager variable at init level so that we can use to delete the temp files created at the end.

Reply:
mhmhmh but we'll need this only on the `connection` files no? we'll create a connection for a service X based on its SSL config. We could have a base SSLManager with basic implementation such as writing the tmp file when needed etc, but keep the method that needs to parse the connection into SSL data as abstract for each service to implement

Reply:
Raised a ticket on enhancement #15913 to tackle above